src/apps/app_bulk_upload.erl

```diff
@@ -0,0 +1,212 @@
+-module(app_bulk_upload).
+-export([upload_file/2, download/2]).
+
+-include("ar.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+%%% Starts a server that accepts possibly huge files, splits them into small chunks, and
+%%% submits them to Arweave as separate transactions. Transactions are submitted via
+%%% the queue module.
+
+-define(CHUNK_SIZE, (1024 * 1024)).
+-define(BLOB_HASH_ALGO, sha256).
+
+%% @doc Starts a queue server, splits the given file into chunks, wraps the chunks as 
+%% Arweave transactions, and submits them to the queue. Returns the queue PID.
+upload_file(Wallet, Filename) ->
+	{ok, Filecontents} = file:read_file(Filename),
+	upload_blob(Wallet, Filecontents).
+
+upload_blob(Wallet, Blob) when is_binary(Blob) ->
+	upload_blob(whereis(http_entrypoint_node), Wallet, Blob).
+
+upload_blob(Node, Wallet, Blob) ->
+	Queue = app_queue:start(Node, Wallet),
+	upload_chunks(Queue, Blob),
+	Queue.
+
+%% @doc Takes a binary blob and processes it chunk by chunk. Each chunk is converted into
+%% a transaction and put into the queue. Chunk size is 1MB.
+upload_chunks(Queue, Blob) ->
+	upload_chunks(Queue, Blob, true).
```

Comment:
`true` and `false` are very inexplicit names for atoms. Let's use something more clear like `first` `not_first`. But, even better than having `first` and `not_first`, let's use a counter starting on 1. What do you think about this?